dynamite_nsm/services/kibana/package/manager.py

```diff
@@ -0,0 +1,784 @@
+    def __init__(self, json_data):
+        super().__init__(json_data, InstalledObjectSchema())
+
+    @staticmethod
+    def from_kwargs(**kwargs):
+        data = {}
+        data['title'] = kwargs.get('title')
+        data['object_type'] = kwargs.get('object_type', None)
+        data['object_id'] = kwargs.get('object_id', None)
+        data['space_id'] = kwargs.get('space_id', None)
+        obj = InstalledObject(data)
+        return obj
+
+    @staticmethod
+    def from_installation_result(responsedata, **kwargs):
+        data = {}
+        data['title'] = responsedata.get('meta', {'title': 'untitled object'}).get('title')
+        data['object_type'] = responsedata.get('type', None)
+        # this is the id of the saved object, not the document for tracking
+        data['object_id'] = responsedata.get('id', None)
+        data.update(kwargs)
+        obj = InstalledObject.from_kwargs(**data)
+        return obj
+
+    def __repr__(self) -> str:
+        return f"<InstalledObect id: {self.object_id}, title: {self.title} >"
+
+class PackageManifest(SchemaToObject):
+    def __init__(self, json_data):
+        """
+        :param json_data: JSON Body of the package manifest, conforms to PackageManifestSchema
+        """
+        super().__init__(json_data, PackageManifestSchema())
+        self.data['slug'] = self.create_slug()
+
+    def create_slug(self):
+        name = self.name or OrphanPackageData.get('name')
+        # using unidecode to support unicode/ascii in the package manifest data.
+        slug = unidecode(name).lower()
+        return re.sub(r'[\W_]+', '-', slug)
+    
+    def json(self) -> str:
+        if not self.data.get('slug', None):
+            self.data['slug'] = self.slug
+        return json.dumps(self.data)
+
+    def __repr__(self) -> str:
+        return f"<PackageManifest(name={self.name}, author={self.author})>"
+
+class Package(object):
+
+    package_index_name = PACKAGES_INDEX_NAME
+    es_url = f'https://{get_primary_ip_address()}:9200'
+    auth = ('admin', 'admin')
+    _installed_objects = []
+    
+    def __init__(self, manifest: PackageManifest,
+                       installed_objects = None,
+                       auth: Optional[str] = None,
+                       id: Optional[str] = None) -> None:
+        self.manifest = manifest
+        if installed_objects:
+            self._installed_objects = installed_objects
+        self.id = id
+        self.auth = auth or Package.auth
+        self.slug = self.manifest.create_slug()
+
+    @staticmethod
+    def es_search(query: dict) -> dict:
+        """Performs an elasticsearch query against the dynamite packages index.
+
+        Args:
+            query (dict): ES DSL for the query
+
+        Raises:
+            ValueError: If response from ES is nonsuccess (not 200 series status)
+
+        Returns:
+            result: dict
+        """
+        result = requests.get(f"{Package.es_url}/{Package.package_index_name}/_search/",
```

Comment:
There's a very good possiblity that a `requests.exceptions.ConnectionError` will be thrown here.

Please log the error, and then raise the exception